setup.py

```diff
@@ -20,48 +20,51 @@
         "https://github.com/DLR-RM/stable-baselines3.git@master"
     )
 
+# pinned to 0.21 until https://github.com/DLR-RM/stable-baselines3/pull/780 goes
+# upstream.
+GYM_VERSION_SPECIFIER = "==0.21.0"
+
+# Note: the versions of the test and doc requirements should be tightly pinned to known
+#   working versions to make our CI/CD pipeline as stable as possible.
 TESTS_REQUIRE = (
     [
-        "seals",
-        "black[jupyter]",
-        "coverage",
-        "codecov",
-        "codespell",
-        "darglint",
-        "filelock",
-        # TODO(adam): remove pin once flake8-isort fixed:
-        #  https://github.com/gforcada/flake8-isort/issues/115
+        "seals==0.1.2",
+        "black[jupyter]~=22.6.0",
+        "coverage~=6.4.2",
+        "codecov~=2.1.12",
+        "codespell~=2.1.0",
+        "darglint~=1.8.1",
+        "filelock~=3.7.1",
         "flake8~=4.0.1",
-        "flake8-blind-except",
-        "flake8-builtins",
-        "flake8-commas",
-        "flake8-debugger",
-        "flake8-docstrings",
-        "flake8-isort",
-        "hypothesis",
-        "ipykernel",
-        "jupyter",
-        # remove pin once https://github.com/jupyter/jupyter_client/issues/637 fixed
```

Comment:
Do we want to lose this comment? Admittedly we don't want to remove the pin any longer, but I think we can move to the 7.x series once it's fixed.

Reply:
You are right. I added it back as a TODO